tools/pre-swap-checker/contracts/PreSwapChecker.sol

```diff
@@ -36,87 +37,96 @@ contract PreSwapChecker {
     wethContract = IWETH(preSwapCheckerWethContract);
   }
 
+
   /**
-    * @notice If order is going through delegate via provideOrder
-    * ensure necessary checks are set
+    * @notice If order is going through wrapper to a delegate
     * @param order Types.Order
     * @param delegate IDelegate
+    * @param wrapper address
     * @return uint256 errorCount if any
     * @return bytes32[] memory array of error messages
     */
-  function checkSwapDelegate(
+  function checkWrapperSwapDelegate(
     Types.Order calldata order,
-    IDelegate delegate
+    IDelegate delegate,
+    address wrapper
     ) external view returns (uint256, bytes32[] memory ) {
-
-    bytes32[] memory errors = new bytes32[](20);
+    bytes32[] memory errors = new bytes32[](22);
     uint256 errorCount;
     address swap = order.signature.validator;
-    IDelegate.Rule memory rule = delegate.rules(order.sender.token,order.signer.token);
-    (uint256 swapErrorCount, bytes32[] memory swapErrors) = checkSwapSwap(order, false);
-
+    (uint256 swapErrorCount, bytes32[] memory swapErrors) = coreSwapChecks(order);
+    (uint256 delegateErrorCount, bytes32[] memory delegateErrors) = coreDelegateChecks(order, delegate);
     if (swapErrorCount > 0) {
       errorCount = swapErrorCount;
       // copies over errors from checkSwapSwap to be outputted
       for (uint256 i = 0; i < swapErrorCount; i++) {
         errors[i] = swapErrors[i];
       }
     }
+    if (delegateErrorCount > 0) {
 
-    // signature must be filled in order to use the Delegate
-    if (order.signature.v == 0) {
-      errors[errorCount] = "SIGNATURE_MUST_BE_SENT";
-      errorCount++;
+      // copies over errors from checkSwapSwap to be outputted
+      for (uint256 i = 0; i < delegateErrorCount; i++) {
+        errors[i + errorCount] = delegateErrors[i];
+      }
+      errorCount = swapErrorCount + delegateErrorCount;
```

Comment:
Can this not just be  `+= delegateErrorCount`? I think that no matter what `swapErrorCount` is, `errorCount == swapErrorCount` is true at this point in time.